vision/sample-apps/V-Start/server.js

```diff
@@ -0,0 +1,230 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const express = require('express');
+const path = require('path');
+const fs = require('fs');
+const fetch = (...args) => import('node-fetch').then(({default: fetch}) => fetch(...args));
+require('dotenv').config();
+
+const app = express();
+const port = process.env.PORT || 8080;
+
+// Middleware & Static File Serving
+app.use(express.json({ limit: '50mb' }));
+app.use(express.static(path.join(__dirname))); // Serves files like index.html, src/, etc.
+
+// --- API ENDPOINTS ---
+
+// 1. Endpoint to securely load the YouTube study data from a private file
+app.get('/api/study/veo-youtube-study', (req, res) => {
+    const filePath = path.join(__dirname, 'data', 'veo-youtube-study.json');
+
+    fs.readFile(filePath, 'utf8', (err, data) => {
+        if (err) {
+            console.error("Error reading study file:", err);
+            return res.status(500).json({ error: "Could not load the study data." });
+        }
+        try {
+            const jsonData = JSON.parse(data);
+            res.json(jsonData);
+        } catch (parseErr) {
+            console.error("Error parsing study JSON:", parseErr);
+            return res.status(500).json({ error: "Study data is corrupted." });
+        }
+    });
+});
+
+// 2. Endpoint to proxy video URLs to avoid CORS issues for user-created studies
+app.get('/api/proxy-video', async (req, res) => {
+    const videoUrl = req.query.url;
+    if (!videoUrl) {
+        return res.status(400).json({ error: 'URL query parameter is required.' });
+    }
+    try {
+        console.log(`Proxying video from: ${videoUrl}`);
+        const videoResponse = await fetch(videoUrl);
+        if (!videoResponse.ok) {
+            throw new Error(`Failed to fetch video with status: ${videoResponse.statusText}`);
+        }
+        const contentType = videoResponse.headers.get('content-type');
+        if (contentType) res.setHeader('Content-Type', contentType);
+        videoResponse.body.pipe(res);
+    } catch (error) {
+        console.error('Error proxying video:', error.message);
+        res.status(500).json({ error: `Failed to proxy video. Reason: ${error.message}` });
+    }
+});
+
+// 3. Token validation endpoint
+app.post('/api/validate-token', async (req, res) => {
+    const { projectId, accessToken } = req.body;
+    
+    console.log('Validating token for project:', projectId);
+    
+    if (!projectId || !accessToken) {
+        return res.status(400).json({ 
+            valid: false, 
+            message: 'Project ID and Token are required.' 
+        });
+    }
+    
+    // Use a simple endpoint to validate the token
+    // We'll try to list models which is a read-only operation
+    const validationUrl = `https://us-central1-aiplatform.googleapis.com/v1/projects/${projectId}/locations/us-central1/models`;
```

Comment:
It's recommended to use the JS Gen AI SDK rather than doing manual HTTP/REST Calls.

https://github.com/googleapis/js-genai

https://cloud.google.com/vertex-ai/generative-ai/docs/sdks/overview#googlegenaisdk_quickstart-nodejs_genai_sdk

Can you please convert this over to use the Gen AI SDK?